sherpa/astro/io/pyfits_backend.py

```diff
@@ -735,7 +735,7 @@ def get_rmf_data(arg, make_copy=False):
         rowdata = []
         for mrow, ng, ncs in zip(matrix, n_grp, n_chan):
             # Need a RMF which ng>1 to test this with.
-            if ng == 1:
+            if numpy.isscalar(ncs):
```

Comment:
So in this case `n_chan = [array([1])]` which means `ncs = array([1])` which means you don't actually want this check to fire, which it would have done, since `ng = 1`.

I guess an alternative approach would be to just say "hey, let's not bother about the input form, let's just always reshape/flatten/whatever the values to the expected shape just before use". This could end up being more lenient, which could be helpful for the wild-west of PHA responses, but I'm not convinced. But I'm also not convinced the current code, which I've only lightly-worked on, is entirely sensible (hence the testing comment).

Reply:
If I understand @DougBurke correctly, this is essentially what I'm doing. Put `[]` around it if it's scalar and don't otherwise. I don't think that's "wile west", though, because the OGIP standard says that this could be a scalar, a fixed-length array, or a variable length array. `numpy.isscalar` will work for those cases; in other words, the OGIP standard tells us to accept scalars or arrays.

Reply:
Yeah, my comment was more a work-through of the code, as I had not understood it originally.

Also, just because the OGIP standard says you can only do a, b, or c doesn't mean some mission doesn't do zeta :-)